### DISCUSSION TITLE
Casks on Linux (AppImage)?

### DISCUSSION THREAD

**Discussion by martin-braun** [Homebrew/discussions#3789] [Linux] [answered]
I really want to open this discussion for a long while and I keep thinking about it, so here we go.

First off, I'm a Mac user, so that's why I choose Homebrew as my primary package manager, obviously.  I love to install Casks as well.

However, I love Linux and I work a lot with it and I've seen some Linux users [praising Homebrew](https://www.youtube.com/watch?v=QsYEvnV-P34), because it allows to get the latest and greatest without touching system dependencies which is kinda similar to what Homebrew does on MacOS.

I was also researching on GUI package formats which got me thinking. AppImage packages for Linux [behave very similar](https://youtu.be/9HuExVD56Bo?t=570) to MacOS's App Package format in which it contains all required dependencies for the application to run and display itself.

It seems there is [no popular package manager](https://askubuntu.com/questions/1062881/do-package-managers-exist-for-appimages) for AppImage packages, people still seem to download and install those packages like MacOS users do with App Packages that aren't using Homebrew.

Seeing all this caused me to ask myself. Why does Homebrew not try to establish a bigger foot in the Linux space by bringing Cask support to Linux by distributing AppImage packages?

I have really much joy with Homebrew and I see myself using it on Linux as well in the future. I think the App Package and AppImage approach is a solid way of distributing full-fledged GUI applications when its backed up by a good package manager, having Homebrew support AppImage Casks on Linux would streamline the full potential on Homebrew across both platforms.

Just a thought, maybe it was discussed in the past and I just [didn't find](https://github.com/Homebrew/discussions/discussions?discussions_q=cask+linux+appimage) the relevant discussion.

**Comment by gromgit** [comment]
As I understand it, cask functionality was introduced in macOS to ease the deployment and maintenance of _closed-source software_, which are a dime a dozen on that platform, not so much on Linux. A central tenet of the Homebrew core repos has always been: if it's open-source, it should be bottled from openly-available source, not from opaque upstream binaries.

**Reply by martin-braun** [reply]
> “To install, drag this icon…” no more!
> 
> Homebrew Cask extends [Homebrew](https://brew.sh/) and brings its elegance, simplicity, and speed to the installation and management of GUI macOS applications such as Atom and Google Chrome.
> 
> We do this by providing a friendly CLI workflow for the administration of macOS applications distributed as binaries.

Source: https://github.com/Homebrew/homebrew-cask

While it's true that more software on Homebrew Casks is closed-source than on Homebrew, this is just a natural coincidence. CLI tools are often open-source, but there are also a lot of Casks that are open-source. Krita, Brave, Firefox, kdenlive, Tor Browser, GIMP, just to name a few.

**Comment by SMillerDev** [comment]
I think the short of it is: make a pull request to add support and then we'll consider it.

**Reply by martin-braun** [reply]
Hey :) Glad you liked my idea so much that you actually brought that feature for us. I agree it would've been better if I would've been involved, but I just started to daily drive Linux. It's such a surprise that you actually did it. Kudos, good job. :+1:

**Comment by dnkmmr69420** [comment]
I was thinking the same thing. Static bins should be on the repo as well.

**Comment by itsTurnip** [comment, marked as answer]
AppImage support [introduced](https://brew.sh/2026/06/11/homebrew-6.0.0/) in Homebrew 6.0

**Reply by itsTurnip** [reply]
Related PR homebrew/brew#20334
